apps/docusaurus/docs/reference/indexer.md

```diff
@@ -0,0 +1,389 @@
+---
+title: "Indexer API Reference"
+---
+
+# TODOs BEFORE MERGING
+
+1. Provide alternatives for deprecated tables if there are any.
+2. Review the descriptions for the tables for accuracy.
+3. Review the types and descriptions for fields for accuracy.
+4. (Optional) Provide an example use case for the table to the description.
+5. (Optional) Change the date on the deprecation notice if we want to. (I guessed for it, assumed all would have the same date, and applied the notice both to the "deprecation" and the "private" fields - so please correct me where I'm wrong!)
+
+Don't worry about the formatting on whitespace for "Description" and such - I'll run a script afterwards before we merge to fix the whitespacing - Jackson
+
+# Indexer API Reference
+
+The Indexer API allows you to access rich data about tokens, accounts, transactions, and events on-chain using GraphQL queries.
+
+You can interactively explore the data discussed here using Hasura explorers here:
+
+- Mainnet: https://cloud.hasura.io/public/graphiql?endpoint=https://api.mainnet.aptoslabs.com/v1/graphql
+- Testnet: https://cloud.hasura.io/public/graphiql?endpoint=https://api.testnet.aptoslabs.com/v1/graphql
+- Devnet: https://cloud.hasura.io/public/graphiql?endpoint=https://api.devnet.aptoslabs.com/v1/graphql
+
+Or via code at the following endpoints:
+
+- Mainnet: https://api.mainnet.aptoslabs.com/v1/graphql
+- Testnet: https://api.testnet.aptoslabs.com/v1/graphql
+- Devnet: https://api.devnet.aptoslabs.com/v1/graphql
+
+As you explore, you can refer back to these reference docs to understand the tables you are interested in. You should use Ctrl + F to quickly find the right entry for that table.
+
+At the bottom of this page we list the tables which are deprecated, along with any alternatives which may replace them. Make sure to check before building any production infrastructure that the table you are using is not deprecated as there was recently a v2 which dramatically simplified the list of tables.
+
+:::tip
+If you are looking up a table with the `_by_pk` suffix, search for the table name without that suffix. `_by_pk` tables are automatically generated for convenience to allow querying by primary key.
+:::
+
+# Indexer Table Reference
+
+:::tip
+Remember to use Ctrl + F to find the table you are interested in! When in doubt, you may also want to query the Hasura tables above to see examples of the data inside.
+:::
+
+## General
```

Comment:
A couple of general comments: 
1. Can we add a primary key column so that we can mark fields as primary keys? 
2. Can we somehow denote the joins separately from the fields? Basically all the Type = Table fields are joins and maybe we can use sorting (fields first, then join keys)? Not sure how you're thinking about it.

Reply:
1. Added all fields that were primary keys. (I verified this by only marking fields which showed up in generated `_by_pk` fields as "required". Please correct me if there are more primary keys than just those. 
2. I moved all the joins to the top of the tables and marked them as "Join" instead of "Table".